corehq/apps/accounting/subscription_changes.py

```diff
@@ -221,6 +222,31 @@ def response_practice_mobile_workers(project, new_plan_version):
         from corehq.apps.app_manager.views.utils import unset_practice_mode_configured_apps
         unset_practice_mode_configured_apps(project.name)
 
+    @staticmethod
+    def response_mobile_worker_creation(domain, new_plan_version):
+        """ deactivates users if there are too many for a community plan """
+        from corehq.apps.accounting.models import (
+            DefaultProductPlan, FeatureType, UNLIMITED_FEATURE_USAGE)
+
+        # checks for community plan
+        if (new_plan_version != DefaultProductPlan.get_default_plan_version()):
+            return
+
+        # checks if unlimited is on for this user
+        user_rate = new_plan_version.feature_rates.filter(
+            feature__feature_type=FeatureType.USER).latest('date_created')
+        if user_rate.monthly_limit == UNLIMITED_FEATURE_USAGE:
+            return
+
+        # checks for extra users
+        num_users = CommCareUser.total_by_domain(
+            domain.name, is_active=True)
+        num_allowed = user_rate.monthly_limit
+        num_extra = num_users - num_allowed
+        if num_extra > 0:
```

Comment:
You could also just do
```python
if num_users > num_allowed:
```

Reply:
ooh, much cleaner. that's what i get for copy pasta